flaky test `test_storage_azure_blob_storage/test_cluster.py::test_unset_skip_unavailable_shards`

`test_storage_azure_blob_storage/test_cluster.py::test_unset_skip_unavailable_shards` is flaky

**Describe the bug**
[Sample report](https://s3.amazonaws.com/clickhouse-test-reports/0/7a5b40563ab6625e2e2722cd8ba3fa282e82037c/integration_tests__release__[4_4].html)

[CI runs](https://play.clickhouse.com/play?user=play#c2VsZWN0IHB1bGxfcmVxdWVzdF9udW1iZXIgYXMgcHIsIGNoZWNrX3N0YXJ0X3RpbWUsIGNoZWNrX25hbWUsIHRlc3RfbmFtZSwgdGVzdF9zdGF0dXMgYXMgc3RhdHVzLCByZXBvcnRfdXJsCmZyb20gY2hlY2tzCndoZXJlIHRvRGF0ZShjaGVja19zdGFydF90aW1lKSA+ICcyMDIzLTA1LTI2JyBhbmQgdGVzdF9uYW1lIGxpa2UgJyV0ZXN0X3Vuc2V0X3NraXBfdW5hdmFpbGFibGVfc2hhcmRzJScgYW5kIHN0YXR1cyBpbiAoJ0ZBSUwnLCAnRVJST1InLCAnRkxBS1knKQpvcmRlciBieSBjaGVja19zdGFydF90aW1lIGRlc2M=)

The error is concerning...
```
E           helpers.client.QueryRuntimeException: Client failed! Return code: 233, stderr: Received exception from server (version 23.12.1):
E           Code: 1001. DB::Exception: Received from 172.16.11.4:9000. DB::Exception: Azure::Storage::StorageException: 404 The specified blob does not exist.
E           
E           Request ID: cd16e9f5-7e05-4f2e-8abd-4e1a32d3c351. Stack trace:
E           
E           0. Azure::Storage::StorageException::CreateFromResponse(std::unique_ptr<Azure::Core::Http::RawResponse, std::default_delete<Azure::Core::Http::RawResponse>>) @ 0x0000000012c6893c in /usr/bin/clickhouse
E           1. Azure::Storage::Blobs::_detail::BlobRestClient::Blob::GetProperties(Azure::Core::Http::_internal::HttpPipeline&, Azure::Core::Url const&, Azure::Storage::Blobs::_detail::BlobRestClient::Blob::GetBlobPropertiesOptions const&, Azure::Core::Context const&) @ 0x0000000012c3943e in /usr/bin/clickhouse
E           2. Azure::Storage::Blobs::BlobClient::GetProperties(Azure::Storage::Blobs::GetBlobPropertiesOptions const&, Azure::Core::Context const&) const @ 0x0000000012c2deea in /usr/bin/clickhouse
E           3. DB::AzureObjectStorage::getObjectMetadata(String const&) const @ 0x000000000fdb0440 in /usr/bin/clickhouse
E           4. DB::StorageAzureBlobSource::KeysIterator::KeysIterator(DB::AzureObjectStorage*, String const&, std::vector<String, std::allocator<String>> const&, std::shared_ptr<DB::IAST>, DB::NamesAndTypesList const&, std::shared_ptr<DB::Context const>, std::vector<DB::RelativePathWithMetadata, std::allocator<DB::RelativePathWithMetadata>>*, std::function<void (DB::FileProgress)>) @ 0x000000001175c5ed in /usr/bin/clickhouse
E           5. DB::StorageAzureBlob::getTableStructureFromData(DB::AzureObjectStorage*, DB::StorageAzureBlob::Configuration const&, std::optional<DB::FormatSettings> const&, std::shared_ptr<DB::Context const>, bool) @ 0x0000000011751286 in /usr/bin/clickhouse
E           6. DB::StorageAzureBlobCluster::StorageAzureBlobCluster(String const&, DB::StorageAzureBlob::Configuration const&, std::unique_ptr<DB::AzureObjectStorage, std::default_delete<DB::AzureObjectStorage>>&&, DB::StorageID const&, DB::ColumnsDescription const&, DB::ConstraintsDescription const&, std::shared_ptr<DB::Context const>, bool) @ 0x000000001176e94c in /usr/bin/clickhouse
E           7. DB::TableFunctionAzureBlobStorageCluster::executeImpl(std::shared_ptr<DB::IAST> const&, std::shared_ptr<DB::Context const>, String const&, DB::ColumnsDescription, bool) const @ 0x000000000f72bc04 in /usr/bin/clickhouse
E           8. DB::ITableFunction::execute(std::shared_ptr<DB::IAST> const&, std::shared_ptr<DB::Context const>, String const&, DB::ColumnsDescription, bool, bool) const @ 0x000000000fc464a3 in /usr/bin/clickhouse
E           9. DB::Context::executeTableFunction(std::shared_ptr<DB::IAST> const&, DB::ASTSelectQuery const*) @ 0x00000000106d534f in /usr/bin/clickhouse
E           10. DB::JoinedTables::getLeftTableStorage() @ 0x000000001108db16 in /usr/bin/clickhouse
E           11. DB::InterpreterSelectQuery::InterpreterSelectQuery(std::shared_ptr<DB::IAST> const&, std::shared_ptr<DB::Context> const&, std::optional<DB::Pipe>, std::shared_ptr<DB::IStorage> const&, DB::SelectQueryOptions const&, std::vector<String, std::allocator<String>> const&, std::shared_ptr<DB::StorageInMemoryMetadata const> const&, std::shared_ptr<DB::PreparedSets>) @ 0x0000000010f866c4 in /usr/bin/clickhouse
E           12. DB::InterpreterSelectWithUnionQuery::InterpreterSelectWithUnionQuery(std::shared_ptr<DB::IAST> const&, std::shared_ptr<DB::Context>, DB::SelectQueryOptions const&, std::vector<String, std::allocator<String>> const&) @ 0x0000000011044a78 in /usr/bin/clickhouse
E           13. DB::InterpreterFactory::get(std::shared_ptr<DB::IAST>&, std::shared_ptr<DB::Context>, DB::SelectQueryOptions const&) @ 0x0000000010f34e3e in /usr/bin/clickhouse
E           14. DB::executeQueryImpl(char const*, char const*, std::shared_ptr<DB::Context>, DB::QueryFlags, DB::QueryProcessingStage::Enum, DB::ReadBuffer*) @ 0x00000000113cacab in /usr/bin/clickhouse
E           15. DB::executeQuery(String const&, std::shared_ptr<DB::Context>, DB::QueryFlags, DB::QueryProcessingStage::Enum) @ 0x00000000113c515a in /usr/bin/clickhouse
E           16. DB::TCPHandler::runImpl() @ 0x00000000122e64e9 in /usr/bin/clickhouse
E           17. DB::TCPHandler::run() @ 0x00000000122fae79 in /usr/bin/clickhouse
E           18. Poco::Net::TCPServerConnection::start() @ 0x0000000014d253d2 in /usr/bin/clickhouse
E           19. Poco::Net::TCPServerDispatcher::run() @ 0x0000000014d261d1 in /usr/bin/clickhouse
E           20. Poco::PooledThread::run() @ 0x0000000014e1cf47 in /usr/bin/clickhouse
E           21. Poco::ThreadImpl::runnableEntry(void*) @ 0x0000000014e1b53c in /usr/bin/clickhouse
E           22. ? @ 0x00007f8555f80ac3 in ?
E           23. ? @ 0x00007f8556012a40 in ?
E           . (STD_EXCEPTION)
E           (query: SELECT count(*) from azureBlobStorageCluster('cluster_non_existent_port','http://azurite1:63225/devstoreaccount1', 'cont', 'test_skip_unavailable.csv', 'devstoreaccount1','Eby8vdM02xNOcqFlqUwJPLlmEtlCDXJ1OUzFT50uSRZ6IFsuFq2UVErCz4I6tq/K1SZFPTOtr/KBHBeksoGMGw=='))
```

--- a/tests/integration/test_storage_azure_blob_storage/test_cluster.py
+++ b/tests/integration/test_storage_azure_blob_storage/test_cluster.py
@@ -203,7 +203,7 @@ def test_unset_skip_unavailable_shards(cluster):
     )
     result = azure_query(
         node,
-        f"SELECT count(*) from azureBlobStorageCluster('cluster_non_existent_port','{storage_account_url}', 'cont', 'test_skip_unavailable.csv', "
+        f"SELECT count(*) from azureBlobStorageCluster('cluster_non_existent_port','{storage_account_url}', 'cont', 'test_unset_skip_unavailable.csv', "
         f"'devstoreaccount1','Eby8vdM02xNOcqFlqUwJPLlmEtlCDXJ1OUzFT50uSRZ6IFsuFq2UVErCz4I6tq/K1SZFPTOtr/KBHBeksoGMGw==')",
     )
 

